File: validate-signature.js
#!/usr/bin/node
let fluree_utils = require('@fluree/crypto-utils');
let fluree_crypto = require('@fluree/crypto-base');
let privkey = "176d072be91c9ab3013435be21af726a84c5f2d22ced506b437a5007ac2cf82d"
let pub_key = fluree_crypto.pub_key_from_private(privkey);
var fs = require('fs');
var obj = JSON.parse(fs.readFileSync(0, 'utf-8'));
var payload = obj["cmd"];
var signature = obj["sig"];
try {
    if (fluree_crypto.verify_signature(pub_key, payload, signature)) {
        console.log("OK");
    } else {
        console.log("FAIL");
    }
} 
catch (err) {
    console.log("FAIL:", err.message)
}

File: validate-signature.py
#!/usr/bin/python3
import sys
import bitcoinlib
import hashlib
import json
import random
import time
import asn1
import base58

class DbSigner:
    def __init__(self, privkey):
        if len(privkey) != 64:
            privkey = base58.b58decode(privkey).hex()
        self.private_key = bitcoinlib.keys.Key(privkey)
        self.public_key = self.private_key.public()
        self.auth_id = self.private_key.address()
        self.decoder =  asn1.Decoder()
    def string_signature_verify(self, datastring, sigstring):
        h = hashlib.sha256()
        h.update(datastring.encode())
        digest = h.digest()
        sigbytes = bytes.fromhex(sigstring)
        self.decoder.start(sigbytes)
        tag, value = self.decoder.read()
        print("DEBUG: signature length", len(value))
        try:
            rval = bitcoinlib.keys.verify(digest, value, self.public_key)
            return rval
        except bitcoinlib.keys.BKeyError as exp:
            print("ERROR from bitcoinlib:", exp)
            return False

def process_stdin(signer):
    obj = json.loads(sys.stdin.read())
    cmd = obj["cmd"]
    sig = obj["sig"]
    if signer.string_signature_verify(cmd, sig):
        print("OK")
    else:
        print("FAIL")


privkey = "bf8a7281f43918a18a3feab41d17e84f93b064c441106cf248307d87f8a60453"
signer = DbSigner(privkey)
process_stdin(signer)
